### PR TITLE
Added validator for rule 387.

### LINKED ISSUE #174: Error code 387 - Reason episode ceased is child moved into independent living arrangement, but the child is aged under 14.
#### Error code
387
#### Description to display
Reason episode ceased is child moved into independent living arrangement, but the child is aged under 14.
#### DfE Long-form description of error
If the reason episode ceased is coded E5 or E6 the age at date episode ceased should be 14th birthday or over.  Check reason episode ceased code and date of birth.
#### DfE suggested coding
If `<REC>` = 'E5' or 'E6' then `<DOB>` must be >`=14 years prior to `<DEC>`

For further info, refer to the DfE validation document [here](https://assets.publishing.service.gov.uk/government/uploads/system/uploads/attachment_data/file/974999/Children_looked_after_data_collection_2020_to_2021_validation_checks_-_version_1.3.pdf)

### PR DESCRIPTION
**Pull request by SLornieCYC**
Closes #174

### CODE REVIEW

File: validator903/validators.py [hunk 1]
```diff
@@ -1,6 +1,40 @@
...
+
+            header['DOB'] = pd.to_datetime(header['DOB'], format='%d/%m/%Y', errors='coerce')
+            episodes['DEC'] = pd.to_datetime(episodes['DEC'], format='%d/%m/%Y', errors='coerce')
+            header['DOB14'] = header['DOB'] + pd.DateOffset(years=14)
+
+            episodes_merged = episodes.reset_index().merge(header, how='left', on=['CHILD'], suffixes=('', '_header'), indicator=True).set_index('index')
+
+            ceased_indep = episodes_merged['REC'].str.upper().astype(str).isin(['E5','E6'])
+            ceased_over_14 = episodes_merged['DOB14'] <= episodes_merged['DEC']
+            dec_present = episodes_merged['DEC'].notna()
```
Comment (dezog) [inline validator903/validators.py:28]: added this small tweak & a couple test cases to make sure missing/invalid DEC's don't trigger this error

**Review by dezog — APPROVED** [review]
Good work, merging now.

Made a small change noted below

File: validator903/validators.py (hunk 1 above)
Comment (SLornieCYC) [reply validator903/validators.py:28]: Thanks! I was relying on `DOB14 <= DEC` returning false if DEC is null (following the to_datetime coerce).

Comment (dezog) [reply validator903/validators.py:28]: Ah - I figured the resulting mask gets flipped with `~`, so they'd end up counting as true. One to watch out for!

**Pull request merged**